src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -890,6 +892,25 @@ private PipelineProcessor CreatePipelineProcessor()
                         }
                     }
 
+                    if (ExperimentalFeature.IsEnabled(ExperimentalFeature.PSNativeCommandPreserveBytePipe))
+                    {
+                        if (commandProcessorBase is NativeCommandProcessor nativeCommandProcessor)
+                        {
+                            nativeCommandProcessor.UpstreamIsNativeCommand = lastCommandWasNative;
+                            if (lastCommandWasNative)
+                            {
+                                lastNativeCommand.DownStreamNativeCommand = nativeCommandProcessor;
+                            }
+
+                            lastCommandWasNative = true;
+                            lastNativeCommand = nativeCommandProcessor;
+                        }
+                        else
+                        {
+                            lastCommandWasNative = false;
+                        }
+                    }
+
                     commandProcessorBase.RedirectShellErrorOutputPipe = this.RedirectShellErrorOutputPipe;
                     pipelineProcessor.Add(commandProcessorBase);
```

Comment:
The same code is now used in `InvokePipeline`, `GetSteppablePipeline`, and `LocalPipeline.CreatePipelineProcessor()`. It would be great if it can be done in `pipelineProcessor.Add`.

Reply:
Done! Much cleaner ty ❤️